client/web/compose/src/store/ui.js

```diff
@@ -0,0 +1,81 @@
+import { compose } from '@cortezaproject/corteza-js'
+
+const types = {
+  loading: 'loading',
+  loaded: 'loaded',
+  pending: 'pending',
+  completed: 'completed',
+  setRecordIds: 'setRecordIds',
+  clearRecordPaginationIds: 'clearRecordPaginationIds',
+}
+
+export default function (ComposeAPI) {
+  return {
+    namespaced: true,
+
+    state: {
+      loading: false,
+      pending: false,
+      recordPaginationIds: [],
+    },
+
+    getters: {
+      loading: (state) => state.loading,
+
+      pending: (state) => state.pending,
+
+      recordPaginationIds (state) {
+        return state.recordPaginationIds
+      },
+    },
+
+    actions: {
+      async loadPaginationRecords ({ commit }, { filter, moduleID, namespaceID, recordListModule, queries, incTotal, incPageNavigation } = {}) {
+        commit(types.pending)
+
+        const [prevRecords, nextRecords] = await Promise.all(queries.map((query) => {
```

Comment:
Instead of awaiting, please return the promise and handle anything that needs to be handled in the `.then` for the Promise.all

Look at any other async action in any another store